app/lib/dns.server.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  Route53Client,
+  CreateHostedZoneCommand,
+  ChangeResourceRecordSetsCommand,
+  GetChangeCommand,
+} from "@aws-sdk/client-route-53";
+import { isIPv4, isIPv6 } from "is-ip";
+
+export const route53Client = new Route53Client({
+  endpoint: process.env.AWS_ENDPOINT_URL || "http://localhost:5053",
+  region: process.env.AWS_REGION || "us-east-1",
+  credentials: {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID || "",
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY || "",
+    sessionToken: process.env.AWS_SESSION_TOKEN,
+  },
+});
+
+export const createHostedZone = async (
```

Comment:
Let's improve the TypeScript for this a bit.  Right now it's giving you untyped responses from the Route53 client. 
 We can fix that.

First, we'll pull in the proper type definition for the `CreateHostedZoneResponse`.  This requires us to `import type ...`.  Then we can type our `response` using `const response: CreateHostedZoneResponse = ...`.  This means that TS knows about the shape and properties of the response, so we can get rid of the `string | undefined` in the return value.

Finally, I've changed the way you were doing the `catch` so that we always return a value or throw.  We'll use proper logging later when we add it in to the server.

Here's how that will look:

```ts
import type { CreateHostedZoneResponse  } from "@aws-sdk/client-route-53";

// ...

export const createHostedZone = async (
  domain: string
) => {
  try {
    const command = new CreateHostedZoneCommand({
      Name: domain,
      CallerReference: new Date().toString(),
    });
    const response: CreateHostedZoneResponse = await route53Client.send(command);

    if (!response.HostedZone?.Id) {
      throw new Error("Missing hosted zone ID in AWS response");
    }
 
    return response.HostedZone.Id.replace("/hostedzone/", "");
  } catch (error) {
    // TODO: Use proper logging...
    console.warn(error);
    throw new Error(`Error while creating hosted zone`);
  }
};
```

We need to follow this pattern for all of the rest below.